src/tests.rs

```diff
@@ -1646,6 +1670,176 @@ fn test_token_introspection_successful_with_basic_auth_full_response() {
+    type TestResult = Result<(), TestError>;
+
+    let result: TestResult = client
+        .set_revocation_url(RevocationUrl::new("http://revocation/url".to_string()).unwrap())
+        .revoke_token(AccessToken::new("access_token_123".to_string()).into())
+        .request(|_| unreachable!());
+
+    match result {
+        Err(RequestTokenError::Other(msg)) => assert_eq!(msg, "revocation_url is not HTTPS"),
+        _ => unreachable!("Expected an error"),
+    };
+}
+
+#[test]
+fn test_token_revocation_with_unsupported_token_type() {
+    let client = new_client()
+        .set_revocation_url(RevocationUrl::new("https://revocation/url".to_string()).unwrap());
+
+    let revocation_response = client
+        .revoke_token(AccessToken::new("access_token_123".to_string()).into())
+        .request(mock_http_client(
+            vec![
+                (ACCEPT, "application/json"),
+                (CONTENT_TYPE, "application/x-www-form-urlencoded"),
+                (AUTHORIZATION, "Basic YWFhOmJiYg=="),
+            ],
+            "token=access_token_123&token_type_hint=access_token",
+            Some("https://revocation/url".parse().unwrap()),
+            HttpResponse {
+                status_code: StatusCode::BAD_REQUEST,
+                headers: vec![(
+                    CONTENT_TYPE,
+                    HeaderValue::from_str("application/json").unwrap(),
+                )]
+                .into_iter()
+                .collect(),
+                body: "{\"error\": \"unsupported_token_type\", \"error_description\": \"stuff happened\", \
+                       \"error_uri\": \"https://errors\"}"
+                    .to_string()
+                    .into_bytes(),
+            },
+        ));
+
+    assert!(matches!(revocation_response, Err(
+        RequestTokenError::ServerResponse(
+            BasicRevocationErrorResponse{
+                error: RevocationErrorResponseType::UnsupportedTokenType,
+                ..
+            })
+        )
+    ));
+}
+
+#[test]
+fn test_token_revocation_with_access_token() {
+    let client = new_client()
+        .set_revocation_url(RevocationUrl::new("https://revocation/url".to_string()).unwrap());
+
+    client
+        .revoke_token(AccessToken::new("access_token_123".to_string()).into())
+        .request(mock_http_client(
+            vec![
+                (ACCEPT, "application/json"),
+                (CONTENT_TYPE, "application/x-www-form-urlencoded"),
+                (AUTHORIZATION, "Basic YWFhOmJiYg=="),
+            ],
+            "token=access_token_123&token_type_hint=access_token",
+            Some("https://revocation/url".parse().unwrap()),
+            HttpResponse {
+                status_code: StatusCode::OK,
+                headers: vec![(
+                    CONTENT_TYPE,
+                    HeaderValue::from_str("application/json").unwrap(),
+                )]
+                .into_iter()
+                .collect(),
+                body: b"{}".to_vec(),
```

Comment:
let's change these response bodies to be empty now that the code handles that correctly

Reply:
Ah, I left it like that because that’s what I saw being returned by Google. Maybe test with both with an actual empty body and empty JSON, and perhaps even with a different response content type?

Reply:
sounds good

Reply:
Done. Perhaps it would be nicer to have less repetition in the tests but verbose tests are obvious and truly isolated and independent of one another. Let me know if you think a helper method or something should be introduced to reduce the (near) repetition in these tests.